src/main/java/edu/hm/hafner/analysis/IssueDifference.java

```diff
@@ -74,19 +106,21 @@ private Issue selectIssueWithSameFingerprint(final Issue current, final List<Iss
     }
 
     private Optional<Issue> findReferenceByFingerprint(final Issue current) {
-        for (Issue reference : fixedIssues) {
-            if (current.getFingerprint().equals(reference.getFingerprint())) {
-                return Optional.of(reference);
-            }
+        List<Issue> references = referencesByFingerprint.get(current.getFingerprint());
+        if (references != null) {
+            return Optional.of(references.get(0));
         }
         return Optional.empty();
     }
 
     private List<Issue> findReferenceByEquals(final Issue current) {
+        List<Issue> references = referencesByHash.get(current.hashCode());
```

Comment:
```suggestion
        List<Issue> references = referencesByHash.getOrDefault(current.hashCode(), new ArrayList<>());
```

Reply:
This would put more pressure on GC creating possibly unneeded object.

Reply:
See #555